get_covid_clade_counts.py

```diff
@@ -52,11 +52,21 @@
 )
 def main(as_of: str | None = None):
     """Get clade counts and save to S3 bucket."""
+    utc_now = datetime.now(tz=timezone.utc)
+    utc_now_date = utc_now.date()
+    if as_of is None or as_of == str(utc_now_date):
+        # as_of not provided or is today
+        as_of_datetime = utc_now
+        as_of = str(utc_now_date)
+    elif as_of > str(utc_now_date):
+        raise ValueError('as_of is in the future!')
+    else:
+        # as_of is not today
+        as_of_datetime = datetime.strptime(as_of, "%Y-%m-%d") \
```

Comment:
This is interesting--do you think it would make more sense for CladeTime to assume dates w/o a time should be set to 23:59:59 instead of midnight?

Reply:
I actually just discussed this with Nick. We both thought that this was what we might expect, but that it is really best to just use a date time if the specific time matters, and neither of us thought this was necessarily worth changing.